src/main/java/org/jenkins/tools/test/hook/TagValidationHook.java

```diff
@@ -17,9 +16,10 @@ public boolean check(@NonNull BeforeCheckoutContext context) {
 
     @Override
     public void action(@NonNull BeforeCheckoutContext context) throws PluginSourcesUnavailableException {
-        Model model = context.getModel();
-        if (model.getScm().getTag() == null || model.getScm().getTag().equals("HEAD")) {
-            throw new PluginSourcesUnavailableException("Failed to check out plugin sources for " + model.getVersion());
+        String tag = context.getPluginMetadata().getTag();
+        if (tag == null || tag.equals("HEAD")) {
```

Comment:
this is incorrect.
we should only abort if we do not know a version in order to update all plugins and checkout out.

the whole point of using the commit hash and not the tag is that so this can work with timestamped snapshots.

As far as the checkout is concerned - using the hash is correct.
if the `hpi-plugin` can not cope with timestamped snapshots when updating dependencies then that is a bug I will need to fix also, however as it stands this regresses support for timestamped snapshots.